nixos/modules/services/backup/mysql-backup.nix

```diff
@@ -76,6 +76,15 @@ in
         chown ${config.services.mysqlBackup.user} ${config.services.mysqlBackup.location}
       '';
 
+    services.mysql.ensureUsers = [{
+      name = config.services.mysqlBackup.user;
+      ensurePermissions = with lib;
+        let
+          privs = "SELECT, SHOW VIEW, TRIGGER, LOCK TABLES";
+          grant = db: nameValuePair "${db}.*" privs;
+        in
+          listToAttrs (map grant config.services.mysqlBackup.databases);
```

Comment:
Didn't think of that use case to reduce permissions to a dynamic set of tables, but I think it's nicely expressed and can serve as a template for other cases where this might be required. 👍